### PR TITLE
WIP es2015, rollup config file, and source maps 

### PR DESCRIPTION
**Pull request by micahstubbs**
This PR shows what an [es2015](https://babeljs.io/learn-es2015/) version of **d3-component** could look like.

new is a `rollup.config.js` file for easier rollup configuration, as well as source maps.

still to do:

- [ ] update tape tests

### CODE REVIEW

**Comment by micahstubbs** [comment]
these links were helpful as I prepared this PR

https://github.com/rollup/rollup-starter-project/blob/master/package.json
https://github.com/rollup/rollup/wiki/Command-Line-Interface
https://github.com/rollup/rollup/wiki/JavaScript-API#globals

**Comment by curran** [comment]
Woohoo! Thanks @micahstubbs this looks awesome. Will take a closer look.

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "url": "https://github.com/curran/d3-component.git"
   },
   "scripts": {
-    "pretest": "rm -rf build && mkdir build && rollup -f umd -g d3-selection:d3 -n d3 -o build/d3-component.js -- index.js",
+    "pretest": "rm -rf build && mkdir build && rollup -c",
```
Comment (curran) [inline package.json]: Nice.

File: package.json [hunk 2]
```diff
@@ -26,7 +26,15 @@
     "jsdom": "^9.11.0",
     "rollup": "0.41",
     "tape": "4",
-    "uglify-js": "2"
+    "uglify-js": "2",
+    "eslint": "^2.9.0",
+    "eslint-config-airbnb": "^9.0.1",
```
Comment (curran) [inline package.json]: Awesome. I'm all for it - ESLint & Babel.

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import babel from 'rollup-plugin-babel';
+import babelrc from 'babelrc-rollup';
+
+let pkg = require('./package.json');
+let external = Object.keys(pkg.dependencies);
+
+export default {
```
Comment (curran) [inline rollup.config.js]: A Good Thing, moving all this to a config file.

File: src/component.js [hunk 4]
```diff
@@ -1,69 +1,78 @@
...
-    getInstance = function (node){ return node.__instance__; },
-    noop = function (){}; // no operation
+import { select } from 'd3-selection';
 
-export default function (tagName, className){
-  var create = noop,
-      render = noop,
-      destroy = noop,
-      key;
+const setInstance = (node, value) => { node.__instance__ = value; }; // no operation
```
Comment (curran) [inline src/component.js]: Oh wow, this stuff cleans up nicely with arrows.

File: src/component.js [hunk 5]
```diff
@@ -1,69 +1,78 @@
...
+    select(this).selectAll('*').each(destroyDescendant);
     (getInstance(this).destroy() || select(this)).remove();
   }
 
-  function destroyDescendant(){
-    var instance = getInstance(this);
-    instance && instance.destroy();
+  function destroyDescendant() {
+    const instance = getInstance(this);
+    if (instance) { instance.destroy(); }
```
Comment (curran) [inline src/component.js]: Yeah this is definitely more clear.

File: src/component.js [hunk 6]
```diff
@@ -1,69 +1,78 @@
...
   }
 
-  component.render = function(_) { return (render = _, component); };
-  component.create = function(_) { return (create = _, component); };
-  component.destroy = function(_) { return (destroy = _, component); };
-  component.key = function(_) { return (key = _, component); };
+  component.render = (_) => {
```
Comment (curran) [inline src/component.js]: I'm guessing this change was ESLint-induced. I'll iterate on this part a bit more. These setters should return `component` for chainability, like this:

```js
component.create = (_) => {
  create = _;
  return component;
};
```

Comment (micahstubbs) [inline src/component.js]: @curran please do improve this.  I actually manually refactored this, in response to an eslint error. I wasn't certain of the return value, and so took a guess. 

the rule it complained about here is http://eslint.org/docs/rules/no-return-assign

**Pull request merged**